alphadia/workflow/optimizer.py

```diff
@@ -0,0 +1,147 @@
+# native imports
+from functools import wraps
+
+import numpy as np
+
+# alpha family imports
+# third party imports
+import pandas as pd
+
+# alphadia imports
+from alphadia.workflow import reporting
+
+
+def check_if_optimum_found(method):
```

Comment:
This is a bit dangerous as we are changing the interface dynamically.

Nevertheless I like the idea with the decorator. Can you change it so it raises a custom exception?
have a look at the exceptions introduced by magnus in this PR: https://github.com/MannLabs/alphadia/pull/279

Reply:
Yes, I will make it return an exception instead. In this case I will remove the decorator from `check_stopping_conditions` (which I will rename `has_converged` as suggested) so that this still returns True if called a subsequent time.